### ISSUE TITLE
Apply-like behaviour that can create or patch existing objects

### ISSUE THREAD

**Issue by ca-scribner** [gtsystem/lightkube#16]
It would be nice to have a function that is similar to `kubectl apply` which lets you both create or (if something already exists) patch the objects via `client.apply(objs)`.  I don't think there's anything that does this natively - do you have this implemented or planned?  If not, do you have interest and an opinion on how it should function?

**Comment by gtsystem** [comment]
Do you have a use case for that? The apply function in `kubectl` is quite [complex](https://github.com/kubernetes/kubectl/tree/master/pkg/apply) as the diff is calculated client side, but we could implement [server side apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/). The go-client does include [that](https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/kubernetes/typed/apps/v1beta1/statefulset.go#L203) and also special [models](https://github.com/kubernetes/client-go/blob/f6ce18ae578c8cca64d14ab9687824d9e1305a67/applyconfigurations/apps/v1/statefulset.go) just to cover this functionality. 

A related interesting [read](https://medium.com/swlh/break-down-kubernetes-server-side-apply-5d59f6a14e26).

**Comment by ca-scribner** [comment]
When writing operators that are trying to ensure `current state == desired state`, it would be great to do:
* render `desired state`
* apply in cluster, updating anything that doesn't match

It is admittedly a bit of a blunt tool, but handy if trying to manage a big bunch of yaml like you'd get from a complex manifest.  `kubectl apply` usually handles that pretty well, and it would be great to have something similar via lightkube.  Atm to do this in lightkube I'd do something like:
```python
for obj in objs:
  try:
    client.create(obj)
  except ApiError as err:
    if err.status == 409:
      client.patch(obj)
    else:
      raise err
```

**Comment by gtsystem** [comment]
It seems that the official kubernetes python client [got support](https://github.com/kubernetes-client/python-base/commit/281f17ab237384bc1f5b022555635710b6e6aff3) for server-side apply (dynamic client only).

**Comment by ca-scribner** [comment]
Sorry I got distracted by other things for a while.  I didn't realise the client now has server-side apply.  Having something like that here would be really nice

**Comment by gtsystem** [comment]
All right, will work on adding support for server-side apply.

**Comment by gtsystem** [comment]
Released as part of v0.10.0

**Comment by ca-scribner** [comment]
Thanks, great work!  We will be putting this to good use in in January
